ui/app/services/api.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service, { service } from '@ember/service';
+import {
+  Configuration,
+  RequestContext,
+  ResponseContext,
+  AuthApi,
+  IdentityApi,
+  SecretsApi,
+  SystemApi,
+} from '@hashicorp/vault-client-typescript';
+import config from '../config/environment';
+
+import type AuthService from 'vault/services/auth';
+import type NamespaceService from 'vault/services/namespace';
+import type ControlGroupService from 'vault/services/control-group';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type { ApiError, ApiResponse, HeaderMap, XVaultHeaders } from 'vault/api';
+
+export default class ApiService extends Service {
+  @service('auth') declare readonly authService: AuthService;
+  @service('namespace') declare readonly namespaceService: NamespaceService;
+  @service declare readonly controlGroup: ControlGroupService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  #responseCache = new Map<string, ApiResponse>();
```

Comment:
I left some comments in the `extractData` middleware but I'm roughing this is for now while I think a bit more on how to handle cases where certain endpoints need access to the outer request data like `wrap_info`.

In short, since the spec does not define a return value (response.data is empty), the client casts the response to `VoidApiResponse` which returns the value as `undefined` regardless of what is returned from the server or the middleware.

Since some of the endpoints in the tools section (which I will be working next) fit into this edge case I'm going to let it sit for now and come back to it.